frontend/src/components/professor/GameManagement/GameLoader/GameLoader.js

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader, Overlay, Spinner } from 'react-bootstrap'
+import JSONInput from 'react-json-editor-ajrm'
+import locale from 'react-json-editor-ajrm/locale/en'
+import { getConfigJson } from './mockData'
+import { debounce } from 'lodash/function'
+import { CustomTooltip } from './GameLoaderStyle'
+
+function GameLoader(props) {
+  const jsonConfig = getConfigJson()
+
+  const [editedJson, setEditedJson] = useState(undefined)
+  const [isSending, setIsSending] = useState(false)
+  const sendButtonRef = useRef()
+  const [showTooltip, setShowTooltip] = useState(false)
+  const [successModalVisible, setSuccessModalVisible] = useState(false)
+
+  const editJsonObject = (e) => {
+    setEditedJson(e.jsObject)
+  }
+
+  const sendJsonConfig = () => {
+    setIsSending(true)
+    if (editedJson) {
+      // TODO: send json data to backend
+      setIsSending(false)
+      setShowTooltip(false)
+      props.setShowModal(false)
+      setSuccessModalVisible(true)
+    } else {
+      setShowTooltip(true)
+      setIsSending(false)
+    }
+  }
+
+  const debounceHideTooltip = useMemo(
+    () =>
+      debounce(() => {
+        setShowTooltip(false)
+      }, 5000),
+    []
+  )
+  useEffect(() => {
+    if (showTooltip) {
+      debounceHideTooltip()
+    }
+  }, [debounceHideTooltip, showTooltip])
+
+  return (
+    <>
+      <Overlay target={sendButtonRef.current} show={showTooltip} placement='bottom'>
+        <CustomTooltip>
+          Niemożliwe jest zapisanie ponieważ nie dokonano żadnych zmian lub zmiany nadal się zapisują. Spróbuj ponownie
+          za chwilę
+        </CustomTooltip>
+      </Overlay>
+
+      <Modal show={props.showModal} onHide={() => props.setShowModal(false)} size={'lg'}>
+        <ModalHeader>
+          <h4 className={'text-center w-100'}>Wczytaj stan gry</h4>
```

Comment:
50-50: może lepiej tak?
```suggestion
          <h4 className={'text-center w-100'}>Wczytaj konfigurację gry</h4>
```